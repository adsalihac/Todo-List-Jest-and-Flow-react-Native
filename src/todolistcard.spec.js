
import React from 'react';
import { FlatList, Text, View, TouchableOpacity } from 'react-native'
import ShallowRenderer from 'react-test-renderer/shallow';
import { findAllWithType, findWithType } from 'react-shallow-testutils'

import TodoListCard from './todolistcard'

import Button from './button'


import type { PropsType } from './todolistcard'
import JSDOMEnvironment from 'jest-environment-jsdom';

const renderer = new ShallowRenderer();

const renderTodoListCard = ({
    onPress,
    name,
    id,
}: PropsType = {} ) => {
    renderer.render(<TodoListCard 
        onPress={onPress}
        name={name}
        id={id}
    />);
    return renderer.getRenderOutput();
}

describe('todolistcard component', () => {
    it('renders correctly', () => {
        const result = renderTodoListCard();
        expect(result).toMatchSnapshot();
    }
    );

    it('renders correctly with props', () => {
        const result = renderTodoListCard({
            onPress: jest.fn(),
            name: 'test',
            id: 1,
        });
        expect(result).toMatchSnapshot();
    });

    

    it('renders a button', () => {
        const result = renderTodoListCard();
        const button = findAllWithType(result, Button);
        expect(button.length).toBe(1);
    }
    );
    



});