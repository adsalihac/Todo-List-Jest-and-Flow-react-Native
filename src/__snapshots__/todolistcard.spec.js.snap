// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todolistcard component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "blue",
      "borderRadius": 5,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "space-between",
      "marginBottom": 10,
      "paddingLeft": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
      }
    }
  />
  <Button
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 5,
        "height": 40,
        "justifyContent": "center",
        "width": 80,
      }
    }
    tilteStyle={
      Object {
        "fontSize": 14,
      }
    }
    title="Delete"
  />
</View>
`;
